src/main/java/sg/ncl/MainController.java

```diff
@@ -1182,14 +1183,35 @@ public String validateExperiment(@ModelAttribute("experimentPageCreateExperiment
         experimentObject.put("idleSwap", "240");
         experimentObject.put("maxDuration", "960");
 
-        System.out.println(experimentObject.toString());
-
         HttpHeaders headers = new HttpHeaders();
         headers.setContentType(MediaType.APPLICATION_JSON);
         headers.set("Authorization", AUTHORIZATION_HEADER);
 
-        HttpEntity<String> request = new HttpEntity<String>(experimentObject.toString(), headers);
-        ResponseEntity responseEntity = restTemplate.exchange(properties.getSioExpUrl(), HttpMethod.POST, request, String.class);
+        HttpEntity<String> request = new HttpEntity<>(experimentObject.toString(), headers);
+        restTemplate.setErrorHandler(new MyResponseErrorHandler());
+        ResponseEntity response = restTemplate.exchange(properties.getSioExpUrl(), HttpMethod.POST, request, String.class);
+
+        String responseBody = response.getBody().toString();
+
+        try {
+            if (RestUtil.isError(response.getStatusCode())) {
+                MyErrorResource error = objectMapper.readValue(responseBody, MyErrorResource.class);
+
+                if (error.getExceptionName().equals(ExceptionState.NSFileParseException.toString())) {
+                    // display error message
+                    redirectAttributes.addFlashAttribute("message", "There is an error when parsing the NS File.");
+                } else if (error.getExceptionName().equals(ExceptionState.ExpNameAlreadyExistsException.toString())) {
+                    // display error message
+                    redirectAttributes.addFlashAttribute("message", "Experiment name already exists.");
+                } else {
+                    // possible sio or adapter connection fail
+                    redirectAttributes.addFlashAttribute("message", "Our server is currently overloaded with requests. Please try again later.");
+                }
+                return "redirect:/experiments/create";
+            }
+        } catch (IOException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1166)